packages/iov-keycontrol/src/userprofile.spec.ts

```diff
@@ -37,6 +42,275 @@ describe("UserProfile", () => {
   const defaultMnemonic3 = "judge edge effort prepare caught lawn mask yellow butter phone dragon more";
   const defaultEncryptionPassword = "my super str0ng and super long password";
 
+  describe("deriveEncryptionKey", () => {
+    it("works", async () => {
+      const key = await UserProfile.deriveEncryptionKey("foobar");
+      expect(key.formatVersion).toEqual(1);
+      expect(key.data.length).toEqual(32);
+    });
+
+    it("can use a different version explicitly", async () => {
+      const key = await UserProfile.deriveEncryptionKey("foobar", 2);
+      expect(key.formatVersion).toEqual(2);
+      expect(key.data.length).toEqual(32);
+    });
+  });
+
+  describe("loadFrom", () => {
+    it("stored in and loaded from storage", async () => {
+      const db = levelup(MemDownConstructor<string, string>());
+
+      const createdAt = new ReadonlyDate("1985-04-12T23:20:50.521Z");
+      const keyring = new Keyring();
+      const original = new UserProfile({ createdAt: createdAt, keyring: keyring });
+
+      await original.storeIn(db, defaultEncryptionPassword);
+
+      const restored = await UserProfile.loadFrom(db, defaultEncryptionPassword);
+
+      expect(restored.createdAt).toEqual(original.createdAt);
+
+      await db.close();
+    });
+
+    it("stored in and loaded from storage when containing special chars", async () => {
+      const db = levelup(MemDownConstructor<string, string>());
+      const wallet1 = Ed25519HdWallet.fromMnemonic(defaultMnemonic3);
+
+      const original = new UserProfile();
+      original.addWallet(wallet1);
+      original.setWalletLabel(wallet1.id, "My secret 😛");
+
+      await original.storeIn(db, defaultEncryptionPassword);
+      const restored = await UserProfile.loadFrom(db, defaultEncryptionPassword);
+
+      expect(restored.wallets.value).toEqual(original.wallets.value);
```

Comment:
It's a bit confusing that different aspects of the profile are being checked, but the titles talk about eg special chars.

Reply:
Indeed. Basically what this test tries to do (and other tests as well) is: perform an equality check on UserProfile (`restored` equals `original`). So I see basically two options to fix this confusion:
1. create a userProfileEquals testing helper that checks all the state that can be checked using public interfaces.
2. only check the very specific data in question (`restored.wallets.value[0].label`)

Reply:
I'm in favour of the helper option.

Reply:
Check, implemented